### PR TITLE
Implement basic authentication

### PR DESCRIPTION
**Pull request by McSinyx**
Resolve GH-27.

### CODE REVIEW

File: src/acanban/auth.py [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+from hmac import compare_digest
+from typing import Dict, Optional, Tuple
+
+from quart import (Blueprint, Quart, ResponseReturnValue,
+                   current_app, redirect, render_template, request)
+from quart_auth import AuthManager, AuthUser, login_user, logout_user
+
+__all__ = ['Authenticator', 'blueprint']
+
+ROLES = 'student', 'supervisor', 'staff', 'admin'
```
Comment (Huy-Ngo) [inline src/acanban/auth.py:29]: I'm going to change "staff" to "assistant" for more descriptive name. I'm going to push that change to a branch as soon as I'm finishing with refactoring the DB.

Comment (McSinyx) [reply src/acanban/auth.py:29]: Do you want me to change it here too?

File: src/acanban/auth.py [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+class Authenticator(AuthManager):
+    user_class = User
+
+    def init_app(self, app: Quart) -> None:
+        """Embed auth_manager attribute into app."""
+        super().init_app(app)
+        # TODO: wrap the database instead
+        self.users: Dict[Optional[str], UserData] = {
+            None: (None, None, None, None),
+            'foo': (crypt('bar'), 'Foo Bar', 'foo@bar.baz', 'student')}
```
Comment (Huy-Ngo) [inline src/acanban/auth.py:75]: Why do we init_app with some non-null values?

Comment (McSinyx) [reply src/acanban/auth.py:75]: For the ease of testing.  I initially wanted to wait until we back user with the db but I think I should remove it now in favor of mock.

**Review by DDHLong — APPROVED** [review]

**Review by Huy-Ngo — APPROVED** [review]

**Comment by McSinyx** [comment]
Thanks y'all!

**Pull request merged**